pkg/process/heartbeat/flusher.go

```diff
@@ -0,0 +1,149 @@
+package heartbeat
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/forwarder"
+	"github.com/DataDog/datadog-agent/pkg/metrics"
+	pstatsd "github.com/DataDog/datadog-agent/pkg/process/statsd"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-go/statsd"
+)
+
+const metricName = "datadog.system_probe.agent.%s"
```

Comment:
Thoughts on moving the logic to create the metric name further up the call stack ? This would not be re-usable in other agents

Reply:
you mean allowing this to be configured?

Reply:
Yes. Or having `Flush` take in a list of metric names instead of a list of modules.